enhancements/hypershift/monitoring.md

```diff
@@ -131,6 +134,28 @@ Metrics exposed by the HyperShift operator itself should be able to answer at le
 
 https://github.com/openshift/hypershift/pull/1376
 
+### Forwarding metrics to guest cluster CMO
+Consumers of HyperShift-managed clusters should be able to observe metrics for their managed control planes in their deployed CMO. We propose 3 options to achieve this integation for HyperShift:
+
+#### Option 1: Compatibility with ROKS toolkit metrics system (sidecar + push gateway)
+A solution based on creating a sidecar container that would scrape metrics from the control plane components (API server, etcd, cluster version operator, etc) and forward that to the guest clusters. 
+This process currently exist within [IBM ROKS toolkit called roks-metrics-deployment](https://github.com/openshift/ibm-roks-toolkit/tree/master/assets/roks-metrics). 
+In short, this is done by injecting a sidecar container on pods already existing on our managed control plane that we would want to collect metrics from. 
+This sidecar container will then push metrics to the push-gateway that will exist on the guest cluster, making control plane metrics available to the guest cluster CMO.
+
+#### Option 2: Collecting and forwarding via Prometheus remote write
+Prometheus remote write features can be utilized to forward control plane metrics from the management cluster to the guest cluster CMO. 
+This approach requires you to have a Prometheus instance available on the management cluster and can be acheived by [configuring remote write](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) 
+to send metrics to the guest cluster CMO's Prometheus [remote write endpoint/receiver](https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver).
+
+#### Option 3: Sidecar and Prometheus remote write (combination of options 1 and 2)
+In this approach, a sidecar container with Prometheus remote write capabilities can be injected to control plane components. 
```

Comment:
I like this option, but I'd like to see some more details fleshed out. 
Will you write custom code to support the remote write protocol or use an existing library? Will it handle multiple shards and retry? (See https://prometheus.io/docs/practices/remote_write/)
How will the sidecar communicate with prometheus in the guest cluster? 
How is a sidecar better than say one additional pod that does this for all servicemonitors/podmonitors in the control plane namespace?
How will you configure remote write for the CMO in the guest cluster? (it's not enabled by default afaik)

Reply:
Also, what happens if the hosted cluster has opted out of running a monitoring operator, or the hosted cluster's monitoring operator is otherwise unavailable (e.g. because of NodePool issues)?  Is that "ah well, customer is allowed to not run monitoring, no need for action"?  Or is there some logic that tries to distinguish "and we expect this hosted cluster to not have a monitoring stack, so we won't bother pushing"?  And in the smart-pusher state, are there management-cluster-side alerts when pushing to the hosted cluster's monitoring stack is expected to work but isn't working?